src/MessagePackAnalyzer/MessagePackCodeFixProvider.cs

```diff
@@ -173,5 +180,20 @@ private static async Task<Solution> AddKeyAttributeAsync(Document document, INam
 
             return solutionEditor.GetChangedSolution();
         }
+
+        private static bool IsPrimaryConstructor(IMethodSymbol constructor, INamedTypeSymbol type)
+        {
+            var constructorParameters = constructor.Parameters;
+            var typeProperties = type.GetMembers()
+                .OfType<IPropertySymbol>()
+                .Where(prop => !prop.IsImplicitlyDeclared)
+                .ToList();
+
+            return constructorParameters.Length == typeProperties.Count &&
```

Comment:
This seems too constrained, as I think it would report `false` for a case like this:

```cs
record A(int B) {
   public int C { get; init; }
   public int D { get; init; }
}
```

Here, the constructor parameters array length is 1, but the type has 3 declared properties (2 of them are not implicit).

Reply:
I deleted this method as it is no longer needed.